packages/client/src/websocket.ts

```diff
@@ -189,9 +189,12 @@ export default class WebSocketClient {
                         console.log('long timeout, or server restart, or sequence number is not found.'); //eslint-disable-line no-console
 
                         this.missedEventCallback?.();
-                        this.missedMessageListeners.forEach((listener) => listener());
 
-                        this.serverSequence = 0;
+                        try {
+                            this.missedMessageListeners.forEach((listener) => listener());
+                        } finally {
+                            this.serverSequence = 0;
+                        }
```

Comment:
FYI I think the code has changed a bit since the related incident. I'm not able to reproduce the spammy behavior with these conditions:

- Throw error in `reconnect`
- Let computer go to sleep so we enter the `computer woke up` loop
- Restart server

The state of the code at that time resulted in pinging the server every 3 seconds, compounding every 30 seconds. I don't see this behavior anymore, with or without the change in this PR.

So it seems the original issue was "fixed" organically. I don't think the `missedMessageListeners` was part of my investigation before. Those listeners could be more involved with the issue than I originally thought.

I would think that the `reconnect` function registered with `addReconnectListener` would be the issue. I see that `reconnect` is called by the `restart` function registered by `addMissedMessageListener`, so it's in the same code path, though the "reason" it gets executed is different than what I would think is the underlying issue.

Reply:
Thx @mickmister for the comments :)

For reproducing, I can constistently reproduce by having an error thrown in reconnect, no need to let the computer go to sleep. Just restarting the server (while having the web app open) is enough to trigger that behaviour (server sequence going out of sync and reconnecting infinitely from the relevant part of the code that calls `this.conn.close`)